### PR TITLE
[SPARK-30358][ML] ML expose predictRaw and predictProbability

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
1, expose predictRaw and predictProbability
2, add tests

### Why are the changes needed?
single instance prediction is useful out of spark, specially for online prediction.
Current `predict` is exposed, but it is not enough.

### Does this PR introduce any user-facing change?
Yes, new methods are exposed

### How was this patch tested?
added testsuites

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala [hunk 1]
```diff
@@ -198,7 +198,8 @@ class DecisionTreeRegressionModel private[ml] (
   }
 
   /** We need to update this function if we ever add other impurity measures. */
-  protected def predictVariance(features: Vector): Double = {
+  @Since("3.0.0")
+  def predictVariance(features: Vector): Double = {
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala]: This is the only one I'm not so sure about. I don't know if it's that useful or if it's well defined ... it's not a variance but some impurity measure, which I suppose happens to be variance in the regressor's prediction at the moment. Not super against it, but is it useful to you?

**Review by srowen — COMMENTED** [review]
OK, I think it's probably reasonable to expose this pretty stable and basic impl

File: mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala (hunk 1 above)
Comment (zhengruifeng) [reply mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala]: You are right. I think we do not need this method for now. I will remove it.

**Comment by zhengruifeng** [comment]
Mereged to master, thanks @srowen for reviewing!

**Pull request closed without merging**